ui/msg/src/view/enhance.ts

```diff
@@ -1,53 +1,6 @@
 import { scroller } from './scroller';
-
-// looks like it has a @mention or #gameid or a url.tld
-export const isMoreThanText = (str: string) => /(\n|(@|#|\.)\w{2,})/.test(str);
-
-export const enhance = (str: string) =>
-  expandGameIds(expandMentions(expandUrls(lichess.escapeHtml(str)))).replace(/\n/g, '<br>');
-
-const expandMentions = (html: string) =>
-  html.replace(/(^|[^\w@#/])@([\w-]{2,})/g, (orig: string, prefix: string, user: string) =>
-    user.length > 20 ? orig : `${prefix}${a('/@/' + user, '@' + user)}`
-  );
-
-const expandGameIds = (html: string) =>
-  html.replace(
-    /\s#([\w]{8})($|[^\w-])/g,
-    (_: string, id: string, suffix: string) => ' ' + a('/' + id, '#' + id, 'text') + suffix
-  );
-
-// ported from https://github.com/bryanwoods/autolink-js/blob/master/autolink.js
-const urlRegex = /(^|[\s\n]|<[A-Za-z]*\/?>)((?:(?:https?|ftp):\/\/|lichess\.org)[\-A-Z0-9+\u0026\u2019@#\/%?=()~_|!:,.;]*[\-A-Z0-9+\u0026@#\/%=~()_|])/gi;
-const expandUrls = (html: string) =>
-  html.replace(urlRegex, (_, space: string, url: string) => `${space}${expandUrl(url)}`);
-
-const expandUrl = (url: string) => expandImgur(url) || expandGiphy(url) || expandImage(url) || expandLink(url);
-
-const imgurRegex = /https?:\/\/(?:i\.)?imgur\.com\/(\w+)(?:\.jpe?g|\.png|\.gif)?/;
-const expandImgur = (url: string) =>
-  imgurRegex.test(url) ? url.replace(imgurRegex, (_, id) => aImg(`https://i.imgur.com/${id}.jpg`)) : undefined;
-
-const giphyRegex = /https:\/\/(?:media\.giphy\.com\/media\/|giphy\.com\/gifs\/(?:\w+-)*)(\w+)(?:\/giphy\.gif)?/;
-const expandGiphy = (url: string) =>
-  giphyRegex.test(url)
-    ? url.replace(giphyRegex, (_, id) => aImg(`https://media.giphy.com/media/${id}/giphy.gif`))
-    : undefined;
-
-const expandImage = (url: string) => (/\.(jpg|jpeg|png|gif)$/.test(url) ? aImg(url) : undefined);
-
-const expandLink = (url: string) => a(url, url.replace(/^https?:\/\//, ''));
-
-const a = (href: string, body: string, cls?: string) =>
-  `<a target="_blank" rel="noopener nofollow" href="${
-    href.startsWith('/') || href.includes('://') ? href : '//' + href
-  }"${cls ? ` class="${cls}"` : ''}>${body}</a>`;
-
-const img = (src: string) => `<img src="${src}"/>`;
-
-const aImg = (src: string) => a(src, img(src));
-
-/* now with the iframe expansion */
+export { isMoreThanText, enhanceAll as enhance } from 'common/richText';
```

Comment:
since these are only used by msg, I think they should remain in msg, so that other modules don't load unnecessary code